scripts/install_moodle

```diff
@@ -0,0 +1,441 @@
+
+  if [ -d "${core_dir_path}" ]; then
+    utils::echo_action "Deleting existing Moodle installation: ${core_dir_path}..."
+    rm -rf "${core_dir_path}"
+    utils::echo_info "Done."
+  fi
+
+  utils::echo_action "Extracting moodle tgz file..."
+  tar zxf "${core_tgz_file}" -C "$(dirname ${core_dir_path})"
+  utils::echo_info "Done."
+}
+
+#######################################
+# Get and extract Moodle plugin files.
+# Arguments:
+#   1) URL of the Moodle plugin zip file to download.
+#   2) MD5 hash code of the Moodle plugin zip file to download.
+#   3) Plugin installation directory path.
+# Outputs:
+#   Writes normal log messages to STDOUT.
+#   Writes error messages to STDERR.
+#######################################
+function get_moodle_plugin() {
+  local plugin_zip_file_url="$1"
+  local plugin_md5_sum="$2"
+  local plugin_dir_path="$3"
+
+  local plugin_zip_file
+
+  utils::echo_action "Downloading plugin zip file..."
+  plugin_zip_file="$(basename ${plugin_zip_file_url})"
+  wget "${plugin_zip_file_url}" -O "${plugin_zip_file}"
+  utils::echo_info "Done."
+
+  utils::echo_action "Checking downloaded file integrity..."
+  if ! echo "${plugin_md5_sum}  ${plugin_zip_file}" | md5sum --check; then
+    utils::echo_error "Downloaded file corrupted. Aborting."
+    exit 1;
+  fi;
+  utils::echo_info "Done."
+
+  utils::echo_action "Extracting plugin files..."
+  unzip -q "${plugin_zip_file}" -d "${plugin_dir_path}"
+  utils::echo_info "Done."
+}
+
+function main() {
+
+  local crontab_entry
+  local moodle_install_options
+  local table_count
+
+  utils::set_exit_trap
+  utils::echo_title "Start of $0"
+
+  #############################################################################
+
+  utils::echo_title "Parse Input Parameters"
+  utils::parse_parameters "$@"
+
+  #############################################################################
+
+  utils::echo_title "Install Tools"
+  utils::echo_action "Installing postgres client, php client and unzip packages..."
+  apt-get install --yes --quiet \
+    php-cli \
+    postgresql-client-10 \
+    unzip
+  utils::echo_info "Done."
+
+  #############################################################################
+
+  utils::echo_title "Install Moodle Dependencies"
+
+  utils::echo_action "Installing apache2 and redis packages..."
+  apt-get install --yes --quiet \
+    apache2 \
+    libapache2-mod-php \
+    redis
+  utils::echo_info "Done."
+
+  utils::echo_action "Installing php packages..."
+  apt-get install --yes --quiet \
+    aspell \
+    clamav \
+    ghostscript \
+    graphviz \
+    php7.2-curl \
+    php7.2-gd \
+    php7.2-intl \
+    php7.2-ldap \
+    php7.2-mbstring \
+    php7.2-pgsql \
+    php7.2-pspell \
+    php7.2-redis \
+    php7.2-soap \
+    php7.2-xml \
+    php7.2-xmlrpc \
+    php7.2-zip
+  utils::echo_info "Done."
+
+  #############################################################################
+
+  utils::echo_title "Setup SMTP Server."
```

Comment:
Period should be removed to keep it consistent

Reply:
Corrigé